trurl: 0.4 -> 0.6, add manpage, add passthru.tests.version

christoph-heiss opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Changelogs:
 - 0.4 -> 0.5: https://github.com/curl/trurl/releases/tag/trurl-0.5
 - 0.5 -> 0.6: https://github.com/curl/trurl/releases/tag/trurl-0.6

This PR also adds a separate `man` output, as well as a basic `passthru.tests.version`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

figsoda commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -1,27 +1,40 @@
 
   enableParallelBuilding = true;
 
+  preBuild = ''
```
```suggestion
  prePatch = ''
```

figsoda commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -1,27 +1,40 @@
+    substituteInPlace version.h --replace 0.4 0.5
+  '';
+
   buildInputs = [ curl ];
```
```suggestion
  nativeBuildInputs = [ curl ];

  buildInputs = [ curl ];

```

christoph-heiss replied:
There is a reason to add `curl` to `nativeBuildInputs` as well? 
Maybe I misunderstood something about `{native,}BuildInputs`, but the trurl build process doesn't need host curl, only for the target.

figsoda replied:
https://github.com/curl/trurl/commit/a7f9ca9c42c2cf29cc84becb38b56082898868d4 switches to using `curl-config`, which should be in `nativeBuildInputs`, you can check this with `strictDeps = true`

christoph-heiss replied:
Oh right, I overlooked, sorry!

figsoda commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -1,27 +1,40 @@
 
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    # Remove wrong, hardcoded version in manpage
+    # https://github.com/curl/trurl/commit/49601415f9fd59b146893c98e47cdd89ebfd070f
+    substituteInPlace trurl.1 --replace "trurl 0.4" trurl
```
please move this to `postPatch`

christoph-heiss replied:
Since I noticed the `installManPage` bit is not needed anyway (as that is already done by the `make install` step), I've decided to move this to prePatch as well, like the `version.h` fixup.

Hopefully that is okay, but happy to change it if the convention says otherwise.

mweinelt reviewed (changes requested):
The manpages should land in a separate `man` output.

figsoda commented on pkgs/tools/networking/trurl/default.nix:
```diff
@@ -1,25 +1,35 @@
 
   doCheck = true;
-  checkInputs = [ perl ];
+  checkInputs = [ python3 ];
```
```suggestion
  nativeCheckInputs = [ python3 ];
```

christoph-heiss committed:
trurl: 0.4 -> 0.6

- upstream switched from using perl to python3 for running tests.
- the build process now relies on `curl-config` being present
- man pages now land in a separate output

Signed-off-by: Christoph Heiss <christoph@c8h4.io>

christoph-heiss committed:
trurl: add passthru.tests.version

Signed-off-by: Christoph Heiss <christoph@c8h4.io>

christoph-heiss commented:
Updated to version [0.6](https://github.com/curl/trurl/releases/tag/trurl-0.6), which was released earlier today.

I also added a simple `passthru.tests.version` test as a separate commit.
